packages/send/backend/src/origins.ts

```diff
@@ -0,0 +1,30 @@
+import { NextFunction, Request, Response } from 'express';
+import { getAllowedOrigins } from './auth/client';
+import cors from 'cors';
+
+const allowedOrigins = getAllowedOrigins();
+
+export const originsHandler = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const origin = req.headers.origin;
+
+  // Check if it's a Thunderbird extension origin
+  if (origin && origin.startsWith('moz-extension://')) {
+    allowedOrigins.push(origin);
+  }
+
+  // Allow any matching origin
+  cors({
+    origin: (origin, callback) => {
+      if (!origin || allowedOrigins.indexOf(origin) !== -1) {
+        callback(null, true);
+      } else {
+        callback(new Error('Origin not allowed by CORS'));
+      }
+    },
+    credentials: true,
+  })(req, res, next);
+};
```

Comment:
Looks good, can we create a middlewares folder and move this in there? There's not a lot of context what this is unless you looked through the code.